### PR TITLE
Sample mobile vitals

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Adds a Mobile Vitals screen in the sample app

### Motivation

Being able to generate different mobile vitals values from the sample app

### CODE REVIEW

File: sample/kotlin/src/main/res/layout/fragment_home.xml [hunk 1]
```diff
@@ -3,130 +3,146 @@
...
+<ScrollView xmlns:android="http://schemas.android.com/apk/res/android"
+            xmlns:app="http://schemas.android.com/apk/res-auto"
+            xmlns:tools="http://schemas.android.com/tools"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent">
+
+    <androidx.constraintlayout.widget.ConstraintLayout android:layout_width="match_parent"
+                                                       android:layout_height="wrap_content"
+                                                       android:padding="10dp"
+                                                       tools:context=".logs.LogsFragment">
```
Comment (0xnm) [inline sample/kotlin/src/main/res/layout/fragment_home.xml:15]: context is wrong

File: sample/kotlin/src/main/res/layout/fragment_home.xml [hunk 2]
```diff
@@ -3,130 +3,146 @@
...
+                android:layout_margin="8dp"
+                android:text="@string/title_about"
+                android:drawableStart="@drawable/ic_info_black_24"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/navigation_picture"/>
+
+
+    </androidx.constraintlayout.widget.ConstraintLayout>
+</ScrollView>
```
Comment (0xnm) [inline sample/kotlin/src/main/res/layout/fragment_home.xml:148]: minor: missing newline at the end of file (same for `fragment_vitals`).

File: sample/kotlin/src/main/res/layout/fragment_vitals.xml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  ~ Copyright 2016-Present Datadog, Inc.
+  -->
+
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+                                                   xmlns:app="http://schemas.android.com/apk/res-auto"
+                                                   xmlns:tools="http://schemas.android.com/tools"
+                                                   android:layout_width="match_parent"
+                                                   android:layout_height="match_parent"
+                                                   android:padding="10dp"
+                                                   tools:context=".logs.LogsFragment">
```
Comment (0xnm) [inline sample/kotlin/src/main/res/layout/fragment_vitals.xml:13]: context should be updated as well.

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsFragment.kt [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        super.onPause()
+        viewModel.pause()
+    }
+
+    // endregion
+
+    // region View.OnClickListener
+
+    override fun onClick(v: View) {
+        requireActivity().runOnUiThread {
```
Comment (0xnm) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsFragment.kt:65]: is it needed? we receive clicks on UI thread anyway.

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+import android.graphics.Bitmap
+import android.os.Handler
+import android.os.Looper
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import java.security.SecureRandom
+
+class VitalsViewModel : ViewModel() {
+
+    val rng = SecureRandom()
```
Comment (0xnm) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsViewModel.kt:17]: minor: seems all fields can be `private`

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        isResumed = false
+        bitmapList.forEach {
+            it.recycle()
+        }
+        bitmapList.clear()
+    }
+
+    fun runLongTask() {
+        val random = SecureRandom()
+        val duration = random.nextInt(250) + 100
```
Comment (0xnm) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsViewModel.kt:55]: why `rng` is not used here? same below.

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsViewModel.kt [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    }
+
+    private fun fillUpMemory() {
+        Thread {
+            loop@ for (i in 0..128) {
+                try {
+                    val bitmap = Bitmap.createBitmap(3840, 2160, Bitmap.Config.ARGB_8888)
+                    bitmapList.add(bitmap)
+                } catch (e: OutOfMemoryError) {
+                    Log.e("Vitals", "Unable to allocte more bitmaps…")
```
Comment (0xnm) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/vitals/VitalsViewModel.kt:102]:
```suggestion
                    Log.e("Vitals", "Unable to allocate more bitmaps…")
```

File: sample/README.md [hunk 8]
```diff
@@ -3,18 +3,48 @@
 ## Getting Started
 
 These sample apps are configured based on configuration JSON files which need to be added in `config` folder in your root directory.
-For each flavor you will have to provide a config file named `[flavorName].json`.
+For each flavor, you must provide a config file named `[flavorName].json`. By default, flavors should match one of the existing site in the `DatadogSite` enum (i.e.: `us1`, `us1_fed`, `us3`, `us5`, `eu1`).
```
Comment (sarina-dd) [inline sample/README.md:6]:
```suggestion
For each flavor, you must provide a config file named `[flavorName].json`. By default, flavors should match one of the existing sites in the `DatadogSite` enum (i.e.: `us1`, `us1_fed`, `us3`, `us5`, `eu1`).
```

File: sample/README.md [hunk 9]
```diff
@@ -3,18 +3,48 @@
...
+```json
+{
+    "apiKey": "YOUR API ID",
+    "applicationKey": "YOUR APPLICATION KEY"
+}
+```
+
+### Staging
+
+If you need to target a site that is not part of the `DatadogSite` enum, you can configure custom endpoints using the following attributes:
```
Comment (sarina-dd) [inline sample/README.md:32]:
```suggestion
If you need to target a site that is not part of the `DatadogSite` enum, configure custom endpoints using the following attributes:
```

**Review by sarina-dd — APPROVED** [review]
Small docs recommendations, otherwise LGTM! 👍

**Review by 0xnm — APPROVED** [review]
nice addition to the sample app

**Pull request merged**